/**:
  ros__parameters:
    prefix: "franka"
    # robot_description: "robot_description"
    planning_group: "fr3_manipulator"
    hand_planning_group: "hand"

    # The rest of these parameters are typical for moveit_cpp
    planning_scene_monitor_options:
      name: "planning_scene_monitor"
      robot_description: "robot_description"
      joint_state_topic: "joint_states"
      attached_collision_object_topic: "attached_collision_object"
      publish_planning_scene_topic: "planning_scene"
      monitored_planning_scene_topic: "monitored_planning_scene"
      wait_for_initial_state_timeout: 10.0

    plan_request_params:
      planner_id: "LIN"
      planning_pipeline: pilz_industrial_motion_planner
      planning_attempts: 5
      planning_time: 10.0
      # 1.0 allow the planner to move the manipulator at full speed
      # which might cause planning failures
      max_velocity_scaling_factor: 0.1
      max_acceleration_scaling_factor: 0.1

    planning_pipelines:
      #namespace: "moveit_cpp"  # optional, default is ~
      pipeline_names: ["ompl", "pilz_industrial_motion_planner"]
      # planning_pipelines:
      # - ompl
      # - pilz_industrial_motion_planner
      # planning_plugin: ompl_interface/OMPLPlanner
      # pipeline_names: ["ompl"]
      namespace: ""

    # ompl ns should be outside of planning_pipelines ns
    ompl:
      fr3_arm:
        planner_configs:
        - SBLkConfigDefault
        - ESTkConfigDefault
        - LBKPIECEkConfigDefault
        - BKPIECEkConfigDefault
        - KPIECEkConfigDefault
        - RRTkConfigDefault
        - RRTConnectkConfigDefault
        - RRTstarkConfigDefault
        - TRRTkConfigDefault
        - PRMkConfigDefault
        - PRMstarkConfigDefault
        - FMTkConfigDefault
        - BFMTkConfigDefault
        - PDSTkConfigDefault
        - STRIDEkConfigDefault
        - BiTRRTkConfigDefault
        - LBTRRTkConfigDefault
        - BiESTkConfigDefault
        - ProjESTkConfigDefault
        - LazyPRMkConfigDefault
        - LazyPRMstarkConfigDefault
        - SPARSkConfigDefault
        - SPARStwokConfigDefault
        - TrajOptDefault
      fr3_manipulator:
        planner_configs:
        - SBLkConfigDefault
        - ESTkConfigDefault
        - LBKPIECEkConfigDefault
        - BKPIECEkConfigDefault
        - KPIECEkConfigDefault
        - RRTkConfigDefault
        - RRTConnectkConfigDefault
        - RRTstarkConfigDefault
        - TRRTkConfigDefault
        - PRMkConfigDefault
        - PRMstarkConfigDefault
        - FMTkConfigDefault
        - BFMTkConfigDefault
        - PDSTkConfigDefault
        - STRIDEkConfigDefault
        - BiTRRTkConfigDefault
        - LBTRRTkConfigDefault
        - BiESTkConfigDefault
        - ProjESTkConfigDefault
        - LazyPRMkConfigDefault
        - LazyPRMstarkConfigDefault
        - SPARSkConfigDefault
        - SPARStwokConfigDefault
        - TrajOptDefault
      hand:
        planner_configs:
        - SBLkConfigDefault
        - ESTkConfigDefault
        - LBKPIECEkConfigDefault
        - BKPIECEkConfigDefault
        - KPIECEkConfigDefault
        - RRTkConfigDefault
        - RRTConnectkConfigDefault
        - RRTstarkConfigDefault
        - TRRTkConfigDefault
        - PRMkConfigDefault
        - PRMstarkConfigDefault
        - FMTkConfigDefault
        - BFMTkConfigDefault
        - PDSTkConfigDefault
        - STRIDEkConfigDefault
        - BiTRRTkConfigDefault
        - LBTRRTkConfigDefault
        - BiESTkConfigDefault
        - ProjESTkConfigDefault
        - LazyPRMkConfigDefault
        - LazyPRMstarkConfigDefault
        - SPARSkConfigDefault
        - SPARStwokConfigDefault
        - TrajOptDefault

      planner_configs:
        BFMTkConfigDefault:
          balanced: 0
          cache_cc: 1
          extended_fmt: 1
          heuristics: 1
          nearest_k: 1
          num_samples: 1000
          optimality: 1
          radius_multiplier: 1.0
          type: geometric::BFMT
        BKPIECEkConfigDefault:
          border_fraction: 0.9
          failed_expansion_score_factor: 0.5
          min_valid_path_fraction: 0.5
          range: 0.0
          type: geometric::BKPIECE
        BiESTkConfigDefault:
          range: 0.0
          type: geometric::BiEST
        BiTRRTkConfigDefault:
          cost_threshold: 1.0e+300
          frountier_node_ratio: 0.1
          frountier_threshold: 0.0
          init_temperature: 100
          range: 0.0
          temp_change_factor: 0.1
          type: geometric::BiTRRT
        ESTkConfigDefault:
          goal_bias: 0.05
          range: 0.0
          type: geometric::EST
        FMTkConfigDefault:
          cache_cc: 1
          extended_fmt: 1
          heuristics: 0
          nearest_k: 1
          num_samples: 1000
          radius_multiplier: 1.1
          type: geometric::FMT
        KPIECEkConfigDefault:
          border_fraction: 0.9
          failed_expansion_score_factor: 0.5
          goal_bias: 0.05
          min_valid_path_fraction: 0.5
          range: 0.0
          type: geometric::KPIECE
        LBKPIECEkConfigDefault:
          border_fraction: 0.9
          min_valid_path_fraction: 0.5
          range: 0.0
          type: geometric::LBKPIECE
        LBTRRTkConfigDefault:
          epsilon: 0.4
          goal_bias: 0.05
          range: 0.0
          type: geometric::LBTRRT
        LazyPRMkConfigDefault:
          range: 0.0
          type: geometric::LazyPRM
        LazyPRMstarkConfigDefault:
          type: geometric::LazyPRMstar
        PDSTkConfigDefault:
          type: geometric::PDST
        PRMkConfigDefault:
          max_nearest_neighbors: 10
          type: geometric::PRM
        PRMstarkConfigDefault:
          type: geometric::PRMstar
        ProjESTkConfigDefault:
          goal_bias: 0.05
          range: 0.0
          type: geometric::ProjEST
        RRTConnectkConfigDefault:
          range: 0.0
          type: geometric::RRTConnect
        RRTkConfigDefault:
          goal_bias: 0.05
          range: 0.0
          type: geometric::RRT
        RRTstarkConfigDefault:
          delay_collision_checking: 1
          goal_bias: 0.1
          range: 0.0
          type: geometric::RRTstar
        SBLkConfigDefault:
          range: 0.0
          type: geometric::SBL
        SPARSkConfigDefault:
          dense_delta_fraction: 0.001
          max_failures: 1000
          sparse_delta_fraction: 0.25
          stretch_factor: 3.0
          type: geometric::SPARS
        SPARStwokConfigDefault:
          dense_delta_fraction: 0.001
          max_failures: 5000
          sparse_delta_fraction: 0.25
          stretch_factor: 3.0
          type: geometric::SPARStwo
        STRIDEkConfigDefault:
          degree: 16
          estimated_dimension: 0.0
          goal_bias: 0.05
          max_degree: 18
          max_pts_per_leaf: 6
          min_degree: 12
          min_valid_path_fraction: 0.2
          range: 0.0
          type: geometric::STRIDE
          use_projected_distance: 0
        TRRTkConfigDefault:
          frountierNodeRatio: 0.1
          frountier_threshold: 0.0
          goal_bias: 0.05
          init_temperature: 1.0e-05
          k_constant: 0.0
          max_states_failed: 10
          min_temperature: 1.0e-09
          range: 0.0
          temp_change_factor: 2.0
          type: geometric::TRRT
        TrajOptDefault:
          type: geometric::TrajOpt
      # one is sufficient
      planning_plugin: ompl_interface/OMPLPlanner
      planning_plugins:
      - ompl_interface/OMPLPlanner
      # can be either a vector of strings or a single string
      request_adapters: >-
        default_planner_request_adapters/ResolveConstraintFrames 
        default_planner_request_adapters/FixWorkspaceBounds
        default_planner_request_adapters/FixStartStateBounds 
        default_planner_request_adapters/FixStartStateCollision
        default_planner_request_adapters/FixStartStatePathConstraints
      # this value is used in fix_start_state_bounds adapter, use the default value for now
      # start_state_max_bounds_error: 0.1

      # can be either a vector of strings or a single string
      response_adapters: >-
        default_planning_response_adapters/ValidateSolution
        default_planning_response_adapters/DisplayMotionPath
      # default_planning_response_adapters/AddTimeOptimalParameterization

    pilz_industrial_motion_planner:
      capabilities: >-
        pilz_industrial_motion_planner/MoveGroupSequenceAction
        pilz_industrial_motion_planner/MoveGroupSequenceService
      default_planner_config: LIN
      planning_plugin: pilz_industrial_motion_planner/CommandPlanner
      planning_plugins:
      - pilz_industrial_motion_planner/CommandPlanner
      request_adapters: >-
        default_planner_request_adapters/ResolveConstraintFrames
        default_planner_request_adapters/FixWorkspaceBounds 
        default_planner_request_adapters/FixStartStateBounds
        default_planner_request_adapters/FixStartStateCollision 
        default_planner_request_adapters/FixStartStatePathConstraints
        default_planner_request_adapters/AddTimeOptimalParameterization
      # this value is used in fix_start_state_bounds adapter, use the default value for now
      # start_state_max_bounds_error: 0.1

      # can be either a vector of strings or a single string
      response_adapters: >-
        default_planning_response_adapters/ValidateSolution
        default_planning_response_adapters/DisplayMotionPath
      # default_planning_response_adapters/AddTimeOptimalParameterization

    # part of pilz_industrial_motion_planner
    robot_description_planning:
      cartesian_limits:
        # max_rot_vel: 1.57
        # max_trans_acc: 2.25
        # max_trans_dec: -5.0
        # max_trans_vel: 1.0
        max_rot_vel: 0.785
        max_trans_acc: 1.125
        max_trans_dec: -2.5
        max_trans_vel: 0.5
      joint_limits:
        fr3_finger_joint1:
          has_acceleration_limits: false
          has_jerk_limits: false
          has_velocity_limits: false
        fr3_finger_joint2:
          has_acceleration_limits: false
          has_jerk_limits: false
          has_velocity_limits: false
        fr3_joint1:
          angle_wraparound: false
          has_acceleration_limits: false
          has_deceleration_limits: false
          has_position_limits: true
          has_jerk_limits: false
          has_velocity_limits: true
          has_effort_limits: true
          has_soft_limits: false
          max_position: 2.7437
          min_position: -2.7437
          max_velocity: 2.62
          max_effort: 87.0
        fr3_joint2:
          angle_wraparound: false
          has_acceleration_limits: false
          has_deceleration_limits: false
          has_position_limits: true
          has_jerk_limits: false
          has_velocity_limits: true
          has_effort_limits: true
          has_soft_limits: false
          max_position: 1.7837
          min_position: -1.7837
          max_velocity: 2.62
          max_effort: 87.0
        fr3_joint3:
          angle_wraparound: false
          has_acceleration_limits: false
          has_deceleration_limits: false
          has_position_limits: true
          has_jerk_limits: false
          has_velocity_limits: true
          has_effort_limits: true
          has_soft_limits: false
          max_position: 2.9007
          min_position: -2.9007
          max_velocity: 2.62
          max_effort: 87.0
        fr3_joint4:
          angle_wraparound: false
          has_acceleration_limits: false
          has_deceleration_limits: false
          has_position_limits: true
          has_jerk_limits: false
          has_velocity_limits: true
          has_effort_limits: true
          has_soft_limits: false
          max_position: -0.1518
          min_position: -3.0421
          max_velocity: 2.62
          max_effort: 87.0
        fr3_joint5:
          angle_wraparound: false
          has_acceleration_limits: false
          has_deceleration_limits: false
          has_position_limits: true
          has_jerk_limits: false
          has_velocity_limits: true
          has_effort_limits: true
          has_soft_limits: false
          max_position: 2.8065
          min_position: -2.8065
          max_velocity: 5.26
          max_effort: 12.0
        fr3_joint6:
          angle_wraparound: false
          has_acceleration_limits: false
          has_deceleration_limits: false
          has_position_limits: true
          has_jerk_limits: false
          has_velocity_limits: true
          has_effort_limits: true
          has_soft_limits: false
          max_position: 4.5169
          min_position: 0.5445
          max_velocity: 4.18
          max_effort: 12.0
        fr3_joint7:
          angle_wraparound: false
          has_acceleration_limits: false
          has_deceleration_limits: false
          has_position_limits: true
          has_jerk_limits: false
          has_velocity_limits: true
          has_effort_limits: true
          has_soft_limits: false
          max_position: 3.0159
          min_position: -3.0159
          max_velocity: 5.26
          max_effort: 12.0

    moveit_controller_manager: moveit_simple_controller_manager/MoveItSimpleControllerManager
    moveit_manage_controllers: true
    moveit_simple_controller_manager:
      controller_names:
      - fr3_arm_controller
      - fr3_gripper
      fr3_arm_controller:
        action_ns: follow_joint_trajectory
        default: true
        joints:
        - fr3_joint1
        - fr3_joint2
        - fr3_joint3
        - fr3_joint4
        - fr3_joint5
        - fr3_joint6
        - fr3_joint7
        type: FollowJointTrajectory
      fr3_gripper:
        action_ns: gripper_action
        default: true
        joints:
        - fr3_finger_joint1
        - fr3_finger_joint2
        type: GripperCommand

    robot_description_kinematics:
      fr3_arm:
        # kinematics_solver: pick_ik/PickIkPlugin
        # kinematics_solver_timeout: 0.05
        # kinematics_solver_attempts: 3
        # mode: local
        # position_scale: 1.0
        # rotation_scale: 0.5
        # position_threshold: 0.001
        # orientation_threshold: 0.01
        # cost_threshold: 0.001
        # minimal_displacement_weight: 0.0
        # gd_step_size: 0.0001
        kinematics_solver: lma_kinematics_plugin/LMAKinematicsPlugin
        kinematics_solver_ik_links:
          - fr3_link8
        kinematics_solver_search_resolution: 0.005
        kinematics_solver_timeout: 0.005
      fr3_manipulator:
        kinematics_solver: lma_kinematics_plugin/LMAKinematicsPlugin
        kinematics_solver_ik_links:
          - fr3_hand_tcp
        kinematics_solver_search_resolution: 0.005
        kinematics_solver_timeout: 0.005
      # hand:
      #   kinematics_solver: null
      #   kinematics_solver_ik_links: null
      #   kinematics_solver_search_resolution: null
      #   kinematics_solver_timeout: null